packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-auth-trigger-template.ts

```diff
@@ -121,3 +140,24 @@ function createPermissionToInvokeLambda(
     sourceArn: userpoolArn.valueAsString,
   });
 }
+
+function createPermissionsForAuthTrigger(
+  stack: cdk.Stack,
+  fnName: cdk.CfnParameter,
+  roleArn: cdk.CfnParameter,
+  permissions: AuthTriggerPermissions,
+  userpoolArn: cdk.CfnParameter,
+) {
+  const myRole = iam.Role.fromRoleArn(stack, 'LambdaExecutionRole', roleArn.valueAsString);
+  new iam.Policy(stack, `${fnName}${permissions.trigger}${permissions.policyName}`, {
+    policyName: permissions.policyName,
+    statements: [
+      new iam.PolicyStatement({
+        effect: permissions.effect === iam.Effect.ALLOW ? iam.Effect.ALLOW : iam.Effect.DENY,
```

Comment:
is permissions.effect is already allow or deny, why do we need this here?

Reply:
These policies are coming from AuthTriggers resource file ```AuthTriggerName.map.json```. Based on this value I am toggling permissions in this file.